src/pipeline_edsl/to_workflow.ml

```diff
@@ -529,14 +667,26 @@ module Make (Config : Compiler_configuration)
         ~name:(sprintf "Input-bam: %s" sample_name)
         ~edges:[depends_on f]
     )
+    |> AF.with_provenance "bam" ["file", AF.get_provenance input]
```

Comment:
So, are we going to use `file` for everything before they fed into to DSL? The way you use these keywords make sense but I can imagine them giving a hard time for somebody who would like to extend this with little knowledge of the framework?

Nothing urgent to address here but maybe we can think about ways to make these more accessible to a newcomer as part of another issue?